### PR TITLE
IA: Check IA reviewer

### PR DESCRIPTION
**Pull request by shargon**
Add automatics reviews to ours prs

### CODE REVIEW

File: .github/workflows/main.yml [hunk 1]
```diff
@@ -49,6 +49,21 @@ jobs:
       with:
         github-token: ${{ secrets.GITHUB_TOKEN }}
 
+  Review:
+    needs: Test
+    runs-on: ubuntu-latest
```
Comment (shargon) [inline .github/workflows/main.yml:54]:
```suggestion
    runs-on: ubuntu-latest
    needs: Test
```

**Comment by cschuchardt88** [comment]
Can't we agree 1st, on which code analyzer to use 1st? Your forcing us into something. How do you even know it solves our problems?

**Comment by shargon** [comment]
> Can't we agree 1st, on which code analyzer to use 1st? Your forcing us into something. How do you even know it solves our problems? Obviously it's wrong and AI analyzers are not that good. There is no model that does a good job. Its trained on every language.

Of course, you can propose a different thing, it is not merged. This is not my idea, @lock9 proposed it in a private community chat.

**Comment by lock9** [comment]
The bot only leaves comments. It doesn't enforce anything. It may not be 100% precise, but it helps.

**Comment by cschuchardt88** [comment]
We don't need something that all it doesn't is complain. What's the point? If we don't know how to program than we shouldn't be working on the project. In addition, doesn't do what we need it to do. `CodeQL` does what we need. Than we have to find a tool to monitor other repos like this one to see if we break changes in it from another repo. We need to get the repos configured and wired up 1st before adding these things. We dont even have an `.editorconfig` in this repo for coding styles and other things.

**Comment by lock9** [comment]
It's different @cschuchardt88,

The bot reviews the algorithm, not only the code. Sometimes, our algorithm has some issues. It may help us to find these issues. 
It's set to use GPT-4, the best AI available.  If the Analyser uses GPT-4 to do code reviews, then I'm fine with it. However, if it doesn't, we need this PR.

**Comment by shargon** [comment]
> We don't need something that all it doesn't is complain. What's the point? If we don't know how to program than we shouldn't be working on the project. In addition, doesn't do what we need it to do. `CodeQL` does what we need. Than we have to find a tool to monitor other repos like this one to see if we break changes in it from another repo. We need to get the repos configured and wired up 1st before adding these things. We dont even have an `.editorconfig` in this repo for coding styles and other things.

Different things, we can have both of them, the bot is like we have a new colaborator how only review the code, you can take his advice or don't, he doesn't approve the pull request, why you don't like it? I can't understand, if someone create a bot in his computer and push the same messages on each commit with a bot, is the same... so please give me your reasons to be against this, and... his will be wrong sometimes is not a valid one, because you and me already are wrong sometimes. Is just a new eyes seeing (and maybe during few days because I dont know how much it will cost me)

**Comment by cschuchardt88** [comment]
> > We don't need something that all it doesn't is complain. What's the point? If we don't know how to program than we shouldn't be working on the project. In addition, doesn't do what we need it to do. `CodeQL` does what we need. Than we have to find a tool to monitor other repos like this one to see if we break changes in it from another repo. We need to get the repos configured and wired up 1st before adding these things. We dont even have an `.editorconfig` in this repo for coding styles and other things.
> 
> Different things, we can have both of them, the bot is like we have a new colaborator who only review the code, you can take his advice or don't, he doesn't approve the pull request, why you don't like it? I can't understand, if someone create a bot in his computer and push the same messages on each commit with a bot, is the same... so please give me your reasons to be against this, and... he will be wrong sometimes is not a valid one, because you and me already are wrong sometimes. Is just a new eyes seeing (and maybe during few days because I dont know how much it will cost me)

What is the bots AI model? It was trained on someone else language and multi-languages and style. Is it fluent in all those languages? NO! Each language is different and it doesn't know style of naming conventions. It doesn't read our `.editorconfig` or other preferences. It doesn't know why something is done. **WE ALL** work on this projects and **WE ALL** need to agree on this.

I am not saying this is bad idea to add bots. But get this repo configured and setup 1st. Before adding a bot. No one takes the time to do anything like that, but me... The bot can point fingers later.

**Comment by cschuchardt88** [comment]
> It's different @cschuchardt88,
> 
> The bot reviews the algorithm, not only the code. Sometimes, our algorithm has some issues. It may help us to find these problems. It's set to use GPT-4, the best AI available. If the analyzer uses GPT-4 to do code reviews, then I'm fine with it. However, if it doesn't, we need this PR.

Are you kidding me? GPT is the worst when it comes to dotnet. I have try many many many many different questions with it. And they all fail. It event told me how to make a nuclear bomb. Dont believe me, i'll share link...discord me.

File: .github/workflows/main.yml [hunk 2]
```diff
@@ -49,9 +49,24 @@ jobs:
...
+        with:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+          OPENAI_API_KEY: ${{ secrets.OPENAI_API_KEY }}
+          OPENAI_API_MODEL: "gpt-4" # Optional: defaults to "gpt-4"
+          exclude: "**/*.json, **/*.md" # Optional: exclude patterns separated by commas
+
   PublishGithub:
-    if: github.ref == 'refs/heads/master' && startsWith(github.repository, 'neo-project/')
     needs: Test
+    if: github.ref == 'refs/heads/master' && startsWith(github.repository, 'neo-project/')
```
Comment (lock9) [inline .github/workflows/main.yml:69]: Why only on the master branch? I think it could be on all branches.

**Comment by Jim8y** [comment]
Hey @cschuchardt88, let's try it for a while, if it's not good, we can remove it.

**Review by cschuchardt88 — APPROVED** [review]

**Review by Jim8y — APPROVED** [review]

File: .github/workflows/main.yml (hunk 2 above)
Comment (shargon) [reply .github/workflows/main.yml:69]: this is for put needs before, not the ia related

**Comment by superboyiii** [comment]
Hard to say. It's good for some suggestion but if somewhere not sure and gpt say it's wrong, it's better to double check than applying directly. Keep in mind GPT is easy to be incorrect in many ways. It just a tool to remind you checking somewhere may have risk.

**Comment by shargon** [comment]
We are trying in neo modules for the moment

**Pull request closed without merging**